salt/proxy/junos.py

```diff
@@ -45,10 +45,24 @@ def init(opts):
     Resource class
     '''
     log.debug('Opening connection to junos')
-    thisproxy['conn'] = jnpr.junos.Device(user=opts['proxy']['username'],
-                                          host=opts['proxy']['host'],
-                                          password=opts['proxy']['passwd'],
-                                          port=opts['proxy']['port'])
+    args = { "host" : opts['proxy']['host'] }
+    optional_args= ['user',
+                    'password'
+                    'port',
+                    'gather_facts', 
+                    'mode',
+                    'baud',
+                    'attempts',
+                    'auto_probe',
+                    'ssh_private_key',
+                    'ssh_config',
+                    'normalize'
+                   ]
+    for arg in optional_args:
+        if arg in opts['proxy'].keys():
```

Comment:
@spidercensus 
In each loop we are calling keys() function. 
can we save it to some variable above for loop start?